asserts/gpgkeypairmgr.go

```diff
@@ -298,7 +298,25 @@ func (gkm *GPGKeypairManager) Export(name string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	return EncodePublicKey(keyInfo.pubKey)
+	return EncodePublicKey(keyInfo.privKey.PublicKey())
+}
+
+// SignWithoutAuthority assembles an assertion without a set authority with the provided information and signs it with the named key.
+func (gkm *GPGKeypairManager) SignWithoutAuthority(assertType *AssertionType, makeHeaders func(pubKey PublicKey) (map[string]interface{}, error), makeBody func(pubKey PublicKey) ([]byte, error), name string) (Assertion, error) {
+	keyInfo, err := gkm.findByName(name)
+	if err != nil {
+		return nil, err
+	}
+	pubKey := keyInfo.privKey.PublicKey()
+	headers, err := makeHeaders(pubKey)
```

Comment:
@cjwatson it's a bit misleading but in our case the PrivateKey passed around is only a reference of the private key, the real key stays within gpg domain